src/storage/index.tsx

```diff
@@ -0,0 +1,120 @@
+import { openDB, IDBPDatabase } from "idb";
+import { Option } from "../util/OptionUtil";
+
+export type HotkeyConfigSettings = unknown;
+export type LayoutSettings = unknown;
+
+const DEFAULT_LAYOUT_WIDTH = 300;
+
+let db: Option<IDBPDatabase<unknown>> = null;
+
+async function getDb(): Promise<IDBPDatabase<unknown>> {
+    if (db == null) {
+        db = await openDB("LiveSplit", 1, {
+            upgrade(db) {
+                db.createObjectStore("splits", {
+                    autoIncrement: true,
+                });
+                db.createObjectStore("layouts", {
+                    autoIncrement: true,
+                });
+                db.createObjectStore("settings", {
+                    autoIncrement: true,
+                });
+            },
```

Comment:
I probably should've told you, but IndexedDB has a notion of migrating between different versions. The migration from no IndexedDB to version 1 is also a migration. This is being done here. And this is the only place we ever need to migrate from localStorage to IndexedDB. So all the migration code should be in here.

Reply:
So would we just load everything from localStorage inside `upgrade` and immediately add them to the database?

Reply:
Yes, I'll push a fix for it in a few minutes.